src/App.tsx

```diff
@@ -15,36 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_LENGTH = 'BY_LENGTH',
+  BY_ALPHABET = 'BY_ALPHABET',
+  DEFAULT = 'DEFAULT',
+}
+
+interface SortOptions {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function preparedList(goods: string[], { sortField, isReversed }: SortOptions) {
+  let prepared = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.BY_ALPHABET:
+        prepared.sort((item1, item2) => item1.localeCompare(item2));
+        break;
+      case SortType.BY_LENGTH:
+        prepared.sort((item1, item2) => item1.length - item2.length);
+        break;
+      case SortType.DEFAULT:
+        prepared = [...goods];
+        break;
+    }
+  }
+
+  if (isReversed) {
+    prepared.reverse();
+  }
+
+  return prepared;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleList = preparedList(goodsFromServer, { sortField, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField === SortType.BY_ALPHABET,
+          })}
+          onClick={() => setSortField(SortType.BY_ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField === SortType.BY_LENGTH,
+          })}
+          onClick={() => setSortField(SortType.BY_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(prevValue => !prevValue)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {sortField !== SortType.DEFAULT && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SortType.DEFAULT);
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleList.map(item => (
+          <li key={item} data-cy="Good">
```

Comment:
[CODE KNOWLEDGE] - When rendering a list, you should provide a unique `key` prop to each child. Using the item as a key is not recommended if the items are not guaranteed to be unique. Consider using the item index or a unique identifier from the item if available.